snapcraft/internal/project_loader/_extensions/gnome_3_34.py

```diff
@@ -0,0 +1,136 @@
+    for GNOME applications.
+
+    It configures each application with the following plugs:
+
+    \b
+    - GTK3 Themes.
+    - Common Icon Themes.
+    - Common Sound Themes.
+    - The GNOME runtime libraries and utilities corresponding to 3.34.
+
+    For easier desktop integration, it also configures each application
+    entry with these additional plugs:
+
+    \b
+    - desktop (https://snapcraft.io/docs/desktop-interface)
+    - desktop-legacy (https://snapcraft.io/docs/desktop-legacy-interface)
+    - gsettings (https://snapcraft.io/docs/gsettings-interface)
+    - wayland (https://snapcraft.io/docs/wayland-interface)
+    - x11 (https://snapcraft.io/docs/x11-interface)
+    """
+
+    @staticmethod
+    def get_supported_bases() -> Tuple[str, ...]:
+        return ("core18",)
+
+    @staticmethod
+    def get_supported_confinement() -> Tuple[str, ...]:
+        return ("strict", "devmode")
+
+    def __init__(self, *, extension_name: str, yaml_data: Dict[str, Any]) -> None:
+        super().__init__(extension_name=extension_name, yaml_data=yaml_data)
+
+        base = yaml_data["base"]
+        platform_snap = _PLATFORM_SNAP[base]
+        self.root_snippet = {
+            "plugs": {
+                "gtk-3-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/themes",
+                    "default-provider": "gtk-common-themes",
+                },
+                "icon-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/icons",
+                    "default-provider": "gtk-common-themes",
+                },
+                "sound-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/sounds",
+                    "default-provider": "gtk-common-themes",
+                },
+                platform_snap: {
+                    "interface": "content",
+                    "target": "$SNAP/gnome-platform",
+                    "default-provider": "{snap}".format(snap=platform_snap),
+                },
+            },
+            "environment": {
+                "SNAP_DESKTOP_RUNTIME": "$SNAP/gnome-platform",
+                "GTK_USE_PORTALS": "1",
+            },
+            "layout": {
+                "/usr/bin/gjs": {"symlink": "$SNAP/gnome-platform/usr/bin/gjs"},
+                "/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0": {
+                    "bind": "$SNAP/gnome-platform/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0"
+                },
+                "/usr/share/xml/iso-codes": {
+                    "bind": "$SNAP/gnome-platform/usr/share/xml/iso-codes"
+                },
+            },
+        }
+
+        self.app_snippet = {
+            "command-chain": ["snap/command-chain/desktop-launch"],
+            "plugs": ["desktop", "desktop-legacy", "gsettings", "wayland", "x11"],
+        }
+
+        self.part_snippet = {
+            "build-environment": [
```

Comment:
why is this not part of the gnome extension part?

Reply:
This would solve the merging issues, the core of snapcraft should do the right thing with "after" and build-environment entries.

Reply:
Because we want the build-environment part snippet to be added to each part, while the gnome-3-34-extension is one whole part that should be added at the part level, and only once.

Reply:
>This would solve the merging issues, the core of snapcraft should do the right thing with "after" and build-environment entries.

Because of the reason listed in my prev comment, I don't see a way to accomplish having the build-environment section added to each part if we combine the part_snippet and part sections.